org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -108,10 +111,31 @@ public void initializeProjects(final Collection<IPath> rootPaths, IProgressMonit
 		} else {
 			importProjectsFromConfigurationFiles(rootPaths, projectConfigurations, monitor);
 		}
+		updateEncoding(monitor);
 		reportProjectsStatus();
 		subMonitor.done();
 	}
 
+	private void updateEncoding(IProgressMonitor monitor) throws CoreException {
+		if (preferenceManager != null && ProjectEncodingMode.SETDEFAULT.equals(preferenceManager.getPreferences().getProjectEncoding())) {
+			String encoding = ResourcesPlugin.getEncoding();
+			for (IProject project : ProjectUtils.getAllProjects()) {
+				if (!Objects.equals(encoding, project.getDefaultCharset(false))) {
+					IWorkspace workspace = ResourcesPlugin.getWorkspace();
+					if (workspace instanceof Workspace) {
+						Workspace ws = (Workspace) workspace;
+						CharsetManager charsetManager = ws.getCharsetManager();
```

Comment:
I would move the lines that work towards computing `charsetManager` outside the `for` loop, just below the declaration of `String encoding = ...` since it's the same value for `CharsetManager` throughout the iterations.

Reply:
Fixed.